src/spikeinterface/exporters/to_ibl.py

```diff
@@ -0,0 +1,309 @@
+            shutil.rmtree(output_folder)
+        else:
+            raise FileExistsError(f"{output_folder} already exists")
+    else:
+        pass
+        # don't make the output dir yet, b/c export_to_phy will do that for us.
+
+    if verbose:
+        print("Exporting recording to IBL format...")
+
+    # Compute any missing extensions
+    available_extension_names = analyzer.get_saved_extension_names()
+    required_exts = [
+        "templates",
+        "template_similarity",
+        "spike_locations",
+        "noise_levels",
+        "quality_metrics",
+    ]
+    required_qms = ["amplitude_median", "isi_violations_ratio", "amplitude_cutoff"]
+    for ext in required_exts:
+        if ext not in available_extension_names:
+            if ext == "quality_metrics":
+                kwargs = {"skip_pc_metrics": not compute_pc_features}
+            else:
+                kwargs = {}
+            analyzer.compute(ext, verbose=verbose, **kwargs)
+        elif ext == "quality_metrics":
+            qm = analyzer.get_extension("quality_metrics").get_data()
+            for rqm in required_qms:
+                if rqm not in qm:
+                    analyzer.compute(
+                        "quality_metrics",
+                        metric_names=[rqm],
+                        verbose=verbose,
+                    )
+
+    # # Start by just exporting to phy
+    if not only_ibl_specific_steps:
+        if verbose:
+            print("Doing phy-like export...")
+        export_to_phy(
+            analyzer,
+            output_folder,
+            compute_amplitudes=compute_amplitudes,
+            compute_pc_features=compute_pc_features,
+            sparsity=sparsity,
+            copy_binary=copy_binary,
+            remove_if_exists=remove_if_exists,
+            template_mode=template_mode,
+            dtype=dtype,
+            verbose=verbose,
+            use_relative_path=use_relative_path,
+            **job_kwargs,
+        )
+
+    if verbose:
+        print("Running IBL-specific steps...")
+
+    # Now we need to add the extra IBL specific files
+    (channel_inds,) = np.isin(analyzer.recording.channel_ids, analyzer.channel_ids).nonzero()
+
+    ### Run spectral density and rms ###
+    fs_ap = analyzer.recording.sampling_frequency
+    rms_win_length_samples_ap = 2 ** np.ceil(np.log2(fs_ap * rms_win_length_sec))
+    total_samples_ap = int(np.min([fs_ap * total_secs, analyzer.recording.get_num_samples()]))
+
+    # the window generator will generates window indices
+    wingen = WindowGenerator(ns=total_samples_ap, nswin=rms_win_length_samples_ap, overlap=0)
+    win = {
+        "TRMS": np.zeros((wingen.nwin, analyzer.recording.get_num_channels())),
+        "nsamples": np.zeros((wingen.nwin,)),
+        "fscale": fscale(welch_win_length_samples, 1 / fs_ap, one_sided=True),
+        "tscale": wingen.tscale(fs=fs_ap),
+    }
+    win["spectral_density"] = np.zeros((len(win["fscale"]), analyzer.recording.get_num_channels()))
```

Comment:
I think we should have a proper function to compute the spectrum if you need it.
So we could have proper tests for this.